aip/client-libraries/4210.md

```diff
@@ -88,6 +88,11 @@ methods for each RPC.
 - Additionally, if the classes generated for each service support using OAuth
   and service credentials, they **must** honor the `google.api.oauth_scopes`
   annotation (if it is provided), and use these scopes by default.
+- Services that have set the `deprecated` protobuf option to `true` **should** have an
+  equivalent deprecation tag generated in the generated class, with a comment
+  specifying when the service will be removed. This is typically the next major
```

Comment:
Why does the service needs to be removed just because it is deprecated?
As an example, in a recent version of the Go language, some APIs have been marked as deprecated however they mention that they'll likely never remove those because it would break the 1.0 backwards compatibility promise.

Is this not a potential valid use-case for RPC APIs as well?

Reply:
Worth mentioning that there may be no hard removal date.

Reply:
Indeed - I suspect there will be plenty of cases where there's a "new and improved" way of doing things, but not enough justification to ever remove the old way.

Reply:
It is worth noting that @miraleung's text here suggests a _version_ rather than a _date_. Realistically, though, it could be either, particularly if some of our deprecation plans go through.

TLDR: This text probably needs to be more open-ended.

Reply:
Updated, PTAL.